src/managed/Microsoft.NET.HostModel/Bundle/Bundler.cs

```diff
@@ -161,63 +124,114 @@ FileType InferType(string fileRelativePath, Stream file)
             return FileType.Extract;
         }
 
-        void GenerateBundle()
+        /// <summary>
+        /// Generate a bundle, given the specification of embedded files
+        /// </summary>
+        /// <param name="fileSpecs">
+        /// An array of tuples representing each file to be embedded:
+        ///   SourcePath: path to the file to be bundled at compile time
+        ///   RelativePath: path where the file is expected at run time, relative to the app DLL.
+        /// </param>
+        /// <returns>
+        /// The full path the the generated bundle file
+        /// </returns>
+        /// <exceptions>
+        /// ArgumentException if input is invalid
+        /// IOExceptions and ArgumentExceptions from callees flow to the caller.
+        /// </exceptions>
```

Comment:
I suggest IEnumerable or IReadOnlyList of tuple here? It can be inefficient to have to make a perfectly sized array and arrays also make the input vs output contract harder to spot since they're mutable

Reply:
I also suggest a custom struct over tuple so that we can add features that require more metadata than relative path with minimal API churn.

Reply:
Thanks @nguerrera. I've changed this now.